src/lib/middleware/api-token-middleware.ts

```diff
@@ -9,21 +14,24 @@ const apiAccessMiddleware = (
     { apiTokenService }: any,
 ): any => {
     const logger = getLogger('/middleware/api-token.ts');
-    logger.info('Enabling api-token middleware');
+    logger.debug('Enabling api-token middleware');
 
     if (!authentication.enableApiToken) {
         return (req, res, next) => next();
     }
 
     return (req, res, next) => {
-        if (req.apiUser) {
+        if (req.user) {
             return next();
         }
 
         try {
             const apiToken = req.header('authorization');
             const apiUser = apiTokenService.getUserForToken(apiToken);
             if (apiUser) {
+                if (apiUser.type === ApiTokenType.CLIENT && !isClientApi(req)) {
+                    return res.sendStatus(403);
+                }
```

Comment:
Nice, and seems reasonable. Calling something a client token, but still allowing access to admin endpoints did seem a bit strange when you pointed it out (even just read access) .